File: advent_of_code/day1/main_test.py
import unittest

from .main import check_sum, cast_to_int, make_pairs, main


class TestDay1(unittest.TestCase):
    def test_check_sum(self) -> None:
        input_data: Tuple[int, int] = 2019, 1
        expected_output: Tuple[int, int, int] = 2019, 1, 2019
        self.assertEqual(check_sum(*input_data), expected_output)

    def test_make_pairs(self) -> None:
        input_data: List[int] = [1, 2, 3]
        expected_output: List[List[int]] = [[1, 2], [1, 3], [2, 3]]
        self.assertEqual(make_pairs(data=input_data), expected_output)
     

File: advent_of_code/day1/main_test.go
package main

import (
	"testing"
)

func TestCheckSum(t *testing.T) {
	input1, input2 := 2019, 1
	expectedOutput := []interface{}{2019, 1, 2019, nil}
	calculatedOutput1, calculatedOutput2, calculatedOutput3, calculatedOutput4 := CheckSum(input1, input2)
	calculatedOutput := []interface{}{calculatedOutput1, calculatedOutput2, calculatedOutput3, calculatedOutput4}

	for i := range calculatedOutput {
		if calculatedOutput[i] != expectedOutput[i] {
			t.Errorf("Output mismatch. Expected %s. Got %s", expectedOutput, calculatedOutput)
			break
		}
	}
}

func TestMakePairs(t *testing.T) {
	// function doesn't currently remove duplicates
	input := []int{1, 2, 3}
	expectedOutput := [][]int{{1, 2}, {1, 3}, {2, 1}, {2, 3}, {3, 1}, {3, 2}}
	calculatedOutput := MakePairs(input)

	for i, val := range calculatedOutput {
		for j := range val {
			if calculatedOutput[i][j] != expectedOutput[i][j] {
				t.Error("Output mismatch", "Expected: ", expectedOutput, "Calculated: ", calculatedOutput)
				break
			}
		}
	}
}
